src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +45,119 @@
+        String currAudience = (String) entity.getProperty("targetAudience");
+        String currInstructions = (String) entity.getProperty("instructions");
+        String currUrl = (String) entity.getProperty("imageUrl");
+        titles.add(currTitle);
+        locations.add(currLocation);
+        startDates.add(currStartDate);
+        endDates.add(currEndDate);
+        contexts.add(currContext);
+        causes.add(currCause);
+        audiences.add(currAudience);
+        instructions.add(currInstructions);
+        urls.add(currUrl);
+    }
+    listOfLists.add(titles);
+    listOfLists.add(locations);
+    listOfLists.add(startDates);
+    listOfLists.add(endDates);
+    listOfLists.add(contexts);
+    listOfLists.add(causes);
+    listOfLists.add(audiences);
+    listOfLists.add(instructions);
+    listOfLists.add(urls);
+    String myJson = convertToJsonWithGson(listOfLists);
+    response.setContentType("application/json;");
+    response.getWriter().println(myJson);
   }
-}
\ No newline at end of file
+
+   private static String convertToJsonWithGson(List<List<String>> arraylist) {
+        Gson myGson = new Gson();
+        String nowJson = myGson.toJson(arraylist);
+        return nowJson;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String initiativeTitle = request.getParameter("initiative-title");
+      String initiativeLocation = request.getParameter("initiative-location");
+      String initiativeStartDate = request.getParameter("initiative-start-date");
+      String initiativeEndDate = request.getParameter("initiative-end-date");
+      String initiativeContext = request.getParameter("initiative-context");
+      String initiativeCause = request.getParameter("initiative-cause");
+      String initiativeTargetAudience = request.getParameter("initiative-target-audience");
+      String initiativeInstructions = request.getParameter("initiative-instructions");
+      String initiativeImageUrl = getUploadedFileUrl(request, "initiative-image");
+      Entity initiativeEntity = new Entity("Initiative");
+      initiativeEntity.setProperty("title", initiativeTitle);
+      initiativeEntity.setProperty("location", initiativeLocation);
+      initiativeEntity.setProperty("startDate", initiativeStartDate);
+      initiativeEntity.setProperty("endDate", initiativeEndDate);
+      initiativeEntity.setProperty("imageUrl", initiativeImageUrl);
+      initiativeEntity.setProperty("context", initiativeContext);
+      initiativeEntity.setProperty("cause", initiativeCause);
+      initiativeEntity.setProperty("targetAudience", initiativeTargetAudience);
+      initiativeEntity.setProperty("instructions", initiativeInstructions);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(initiativeEntity);
+      response.sendRedirect("/createYourOwn.html");
```

Comment:
Perhaps it's a better idea to redirect the user to their home page, profile page, or another page that says "Initiative created successfully," but that can be something to consider later.